rust/db_handling/src/identities.rs

```diff
@@ -1104,3 +1214,211 @@ pub fn prepare_derivations_import(
         &derivations,
     ))
 }
+
+#[cfg(feature = "signer")]
+pub(crate) fn prepare_secret_key_for_export(
+    multisigner: &MultiSigner,
+    full_address: &str,
+    pwd: Option<&str>,
+) -> Result<[u8; 32]> {
+    match multisigner {
+        MultiSigner::Ed25519(public) => {
+            let ed25519_pair = match ed25519::Pair::from_string(full_address, pwd) {
+                Ok(x) => x,
+                Err(e) => return Err(Error::SecretStringError(e)),
+            };
+            if public != &ed25519_pair.public() {
+                return Err(Error::WrongPassword);
+            }
+            Ok(*ed25519_pair.seed())
+        }
+        MultiSigner::Sr25519(public) => {
+            let (sr25519_pair, seed) = match sr25519::Pair::from_string_with_seed(full_address, pwd)
+            {
+                Ok(x) => x,
+                Err(e) => return Err(Error::SecretStringError(e)),
+            };
```

Comment:
I believe this can be the same as in the other two branches and call `from_string`

Reply:
It's problematic. `sr25519::Pair::from_string` returns a `Pair`, that doesn't have `seed` method like others. I could use `to_raw_vec()` to get `Vec<u8>`, but for sr25519 it returns 64 bytes key and it's not recognised by `polkadot.js` extention.